### PR TITLE
[5.1] Add Fanout event broadcasting

### PR DESCRIPTION
**Pull request by cwt137**
Fanout is a real-time, cloud based event system like Pusher. http://fanout.io . It would be nice if Laravel supported this system as well.

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
Hmmmm. I don't think we need to include everything in the core.

**Comment by cwt137** [comment]
Just thought it would cool to have a nice list of options. For example the Queue service has a nice list including IronMQ, Amazon SQS, Beanstalkd, Redis (Resque based?) and more.

**Comment by lukasgeiter** [comment]
I guess if #8685 gets merged it would be only "fair" to add this too :smile:

**Comment by GrahamCampbell** [comment]
An example of where we have few drivers in the core is socialite. There's over 50 other community drivers for it, and less than 10 core drivers. I don't think we should have more than 5 drivers in the core, so we should chose them carefully. :)

File: composer.json [hunk 1]
```diff
@@ -104,6 +104,7 @@
     "suggest": {
         "aws/aws-sdk-php": "Required to use the SQS queue driver and SES mail driver (~2.4).",
         "doctrine/dbal": "Required to rename columns and drop SQLite columns (~2.4).",
+        "fanout/fanout": "Required to use the Fanout broadcast driver (~2.0).",
```
Comment (GrahamCampbell) [inline composer.json]: You forgot about the other composer.json file.

Comment (cwt137) [inline composer.json]: I put that line in both the composer.json file and src/Illuminate/Broadcasting/composer.json. Is there another file I was supposed to put it into?

**Comment by ConnorVG** [comment]
Seems like redis, pusher and pubnub are a strong suite for core drivers. I understand why people want more but it seems a bit bloat-like for this specific feature. I agree with @GrahamCampbell on this one :)

**Comment by jkarneges** [comment]
If anyone is interested in this, we've created a separate package outside of the core: https://github.com/fanout/laravel-fanout

**Pull request closed without merging**